lib/appsignal/check_in/debouncer.rb

```diff
@@ -0,0 +1,190 @@
+# frozen_string_literal: true
+
+module Appsignal
+  module CheckIn
+    class Debouncer
+      SHORT_DEBOUNCE_SECONDS = 0.1
+      LONG_DEBOUNCE_SECONDS = 10
+
+      def initialize
+        # The mutex is used to synchronize access to the events array, the
+        # waker thread and the main thread, as well as queue writes
+        # (which depend on the events array) and closes (so they do not
+        # happen at the same time that an event is added to the debouncer)
+        @mutex = Mutex.new
+        # The thread will be started when an event is first added.
+        @thread = nil
+        @queue = Thread::Queue.new
+        @events = []
+        # The waker thread is used to schedule debounces. It will be started
+        # when an event is first added.
+        @waker = nil
+        # For internal testing purposes.
+        @transmitted = 0
+      end
+
+      def send(event)
```

Comment:
I would rename this method so you're not overwriting Ruby's `send` method. Like `transmit` and rename the private `transmit` method.

Reply:
Done! Went with `schedule` instead.